opensearch/src/test/java/org/opensearch/sql/opensearch/executor/OpenSearchExecutionEngineTest.java

```diff
@@ -252,4 +285,42 @@ public String explain() {
       return "explain";
     }
   }
+
+  public static class FakeOpenSearchIndexScan extends OpenSearchIndexScan {
```

Comment:
Feels like you could `@Mock` this class and use the `when` override instead of declaring these functions protected with an `@Override`.  But maybe you tried that and it didn't work out...?

Reply:
Gave it another shot and it worked. Changed in https://github.com/Bit-Quill/opensearch-project-sql/pull/217/commits/c7208ea1f35ee9d657fb6898e800edb77e142ab3